File: scripts/upload-s3.py
#!/usr/bin/env python3
import sys
import os
import boto3
from botocore.exceptions import ClientError

def upload_to_s3(bucket, key, content):
    region = os.environ.get('AWS_REGION', 'prod')
    
    access_key = os.environ.get('BOSS_ACCESS_KEY_ID') or os.environ.get('AWS_ACCESS_KEY_ID')
    secret_key = os.environ.get('BOSS_SECRET_ACCESS_KEY') or os.environ.get('AWS_SECRET_ACCESS_KEY')
    endpoint = os.environ.get('BOSS_ENDPOINT')
    
    if not access_key or not secret_key:
        print("Error: Missing BOSS_ACCESS_KEY_ID or BOSS_SECRET_ACCESS_KEY", file=sys.stderr)
        sys.exit(1)
    
    config = {
        'region_name': region,
        'aws_access_key_id': access_key,
        'aws_secret_access_key': secret_key,
    }
    
    if endpoint:
        config['endpoint_url'] = endpoint
    
    s3_client = boto3.client('s3', **config)
    
    try:
        extra_args = {'ContentType': 'text/html; charset=utf-8'}
        if endpoint:
            extra_args['ACL'] = 'public-read'
        
        s3_client.put_object(
            Bucket=bucket,
            Key=key,
            Body=content,
            **extra_args
        )
        # 生成URL
        if endpoint:
            url = f"{endpoint}/{bucket}/{key}"
        else:
            url = f"https://{bucket}.s3.{region}.amazonaws.com/{key}"
        print(url)
    except ClientError as e:
        print(f"Error uploading: {e}", file=sys.stderr)
        sys.exit(1)

if __name__ == '__main__':
    bucket = sys.argv[1] if len(sys.argv) > 1 else 'copilot'
    key = sys.argv[2] if len(sys.argv) > 2 else None
    
    if not key:
        print("Usage: upload-s3.py <bucket> <key>", file=sys.stderr)
        sys.exit(1)
    
    # 读取标准输入
    content = sys.stdin.buffer.read()
    upload_to_s3(bucket, key, content)

File: scripts/upload-s3.mjs
#!/usr/bin/env node
import { S3Client, PutObjectCommand } from "@aws-sdk/client-s3";
// 直接使用 process.env 代替 tiny-getenv

const bucket = process.argv[2] || "copilot";
const key = process.argv[3];
const region = process.env.AWS_REGION || "prod";

if (!key) {
  console.error("Usage: node upload-s3.mjs <bucket> <key>");
  process.exit(1);
}

const accessKeyId = process.env.BOSS_ACCESS_KEY_ID || process.env.AWS_ACCESS_KEY_ID;
const secretAccessKey = process.env.BOSS_SECRET_ACCESS_KEY || process.env.AWS_SECRET_ACCESS_KEY;
const endpoint = process.env.BOSS_ENDPOINT;

if (!accessKeyId || !secretAccessKey) {
  console.error("Error: Missing BOSS_ACCESS_KEY_ID or BOSS_SECRET_ACCESS_KEY");
  process.exit(1);
}

const s3 = new S3Client({
  region,
  credentials: { accessKeyId, secretAccessKey },
  ...(endpoint && {
    endpoint,
    forcePathStyle: true, // 强制使用路径样式
  }),
});

let body = "";
process.stdin.setEncoding("utf8");
for await (const chunk of process.stdin) {
  body += chunk;
}

const command = new PutObjectCommand({
  Bucket: bucket,
  Key: key,
  Body: body,
  ContentType: "text/html; charset=utf-8",
});

await s3.send(command);
// 对于内网BOSS，使用固定路径格式
const url = endpoint
  ? `https://shjd-inner-boss.bilibili.co/copilot/${key}`
  : `https://${bucket}.s3.${region}.amazonaws.com/${key}`;
console.log(url);
